topk-js/src/schema/mod.rs

```diff
@@ -79,11 +90,16 @@ pub struct SemanticIndexOptions {
 }
 
 #[napi(namespace = "schema")]
-pub fn semantic_index(options: SemanticIndexOptions) -> FieldIndex {
+pub fn semantic_index(options: Option<SemanticIndexOptions>) -> FieldIndex {
+    let (model, embedding_type) = match options {
+        Some(options) => (options.model, options.embedding_type),
+        None => (None, None),
+    };
+
     FieldIndex {
         index: Some(FieldIndexUnion::SemanticIndex {
-            model: options.model,
-            embedding_type: options.embedding_type,
+            model,
```

Comment:
nit: you can write this as:

```js
FieldIndexUnion::SemanticIndex {
    model: options.map(|o| o.model),
    embedding_type: options.map(|o| o.embedding_type),
}
```

which is "the rust way" to do it

Reply:
I'd need to unwrap tho:
![Screenshot 2025-04-23 at 19 53 50](https://github.com/user-attachments/assets/9e10ad4f-e992-4e51-abea-2c6394198e5e)

And then:
![Screenshot 2025-04-23 at 19 54 32](https://github.com/user-attachments/assets/37b9f0ff-3e2e-4267-8999-5460a7db08f4)

Reply:
to get rid of `Option<Option<String>>` you can do `options.map(|o| o.model).flatten()`. but yeah you'd need `options.as_ref().map(|o| o.model.clone()).flatten()`